network/p2p/middleware.go

```diff
@@ -367,7 +387,9 @@ func (m *Middleware) Subscribe(channel network.Channel) error {
 	go rs.receiveLoop(m.wg)
 
 	// update peers to add some nodes interested in the same topic as direct peers
-	m.peerManager.RequestPeerUpdate()
+	if m.managePeerConnections {
+		m.peerManager.RequestPeerUpdate()
```

Comment:
Nit: that works, but there's a naked `nil` `peerManager` in the middleware that's a crash waiting to happen if it's one day accessed without checking the flag. Could we use a getter here?